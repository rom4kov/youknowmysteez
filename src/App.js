import { useEffect } from "react";
import { Routes, Route } from "react-router-dom";
import { useDispatch } from "react-redux";
import { useInView } from "react-intersection-observer";
import Navigation from "./routes/navigation/navigation.component";
import Home from "./routes/home/home.component";
import Authentication from "./routes/authentication/authentication.component";
import Shop from "./routes/shop/shop.component";
import Checkout from "./routes/checkout/checkout.component";
import { checkUserSession } from "./store/reducers/user.reducer";

const App = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    console.log(checkUserSession());
    dispatch(checkUserSession())
  }, [dispatch]);

  let { ref: intObsRef, inView: navColor } = useInView({
    threshold: 0.2,
  });

  return (
    <Routes>
      <Route path="/" element={<Navigation navColor={navColor} />}>
        <Route index element={<Home intObsRef={intObsRef} />} />
        <Route path="auth" element={<Authentication />} />
        <Route path="shop/*" element={<Shop />} />
        <Route path="checkout" element={<Checkout />} />
      </Route>
    </Routes>
  );
};

export default App;
