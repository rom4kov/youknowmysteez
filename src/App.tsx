import { useEffect } from "react";
import { Routes, Route } from "react-router-dom";
import { useDispatch } from "react-redux";
import { useInView } from "react-intersection-observer";
import Navigation from "./routes/navigation/navigation.component";
import Home from "./routes/home/home.component";
import Authentication from "./routes/authentication/authentication.component";
import Shop from "./routes/shop/shop.component";
import Cart from "./routes/cart/cart.component";
import Payment from "./routes/payment/payment.component";
import Confirmation from "./routes/payment/confirmation.component";
import { checkUserSession } from "./store/reducers/user.reducer";

const App = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(checkUserSession({}));
  }, [dispatch]);

  let { ref: intObsRef, inView: navColor } = useInView({
    threshold: 0.2,
  });

  console.log(intObsRef);

  return (
    <Routes>
      <Route path="/" element={<Navigation navColor={navColor} />}>
        <Route index element={<Home intObsRef={intObsRef} />} />
        <Route path="auth" element={<Authentication />} />
        <Route path="shop/*" element={<Shop />} />
        <Route path="cart" element={<Cart />} />
        <Route path="payment" element={<Payment />} />
        <Route path="confirmation" element={<Confirmation />} />
      </Route>
    </Routes>
  );
};

export default App;
