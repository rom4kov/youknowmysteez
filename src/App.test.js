import { act, mount } from "enzyme";
import App from "./App";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";

const mockStore = configureStore();
const initState = {};
const store = mockStore(initState);

test("renders learn react link", () => {
  expect(
    mount(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>
    )
  ).toMatchSnapshot();
});
