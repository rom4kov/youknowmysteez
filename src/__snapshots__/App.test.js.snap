// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      location={
        Object {
          "hash": "",
          "key": "default",
          "pathname": "/",
          "search": "",
          "state": null,
        }
      }
      navigationType="POP"
      navigator={
        Object {
          "action": "POP",
          "createHref": [Function],
          "createURL": [Function],
          "encodeLocation": [Function],
          "go": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "default",
            "pathname": "/",
            "search": "",
            "state": null,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <App>
        <Suspense
          fallback={<Spinner />}
        >
          <Spinner>
            <styled.div>
              <div
                className="sc-beqWaB calAUi"
              >
                <styled.div>
                  <div
                    className="sc-gueYoa jQLknU"
                  />
                </styled.div>
              </div>
            </styled.div>
          </Spinner>
        </Suspense>
      </App>
    </Router>
  </BrowserRouter>
</Provider>
`;
